apps/credential-showcase-api-server/src/database/repositories/ScenarioRepository.ts

```diff
@@ -552,35 +564,41 @@ class ScenarioRepository implements RepositoryDefinition<Scenario, NewScenario>
         })
         .returning()
 
-      const actionsResult = await tx
-        .insert(stepActions)
-        .values(
-          step.actions.map((action: NewAriesOOBAction) => ({
+      if (step.actions && step.actions.length > 0) {
+        const actionsResult = await tx
+          .insert(stepActions)
+          .values(
+            step.actions.map((action: NewAriesOOBAction) => ({
+              ...action,
+              step: stepResult.id,
+            })),
+          )
+          .returning()
+
+        const proofRequestsResult = await tx
+          .insert(ariesProofRequests)
+          .values(
+            step.actions.map((action, index) => {
+              const stepAction = actionsResult[index]
+              return {
+                ...action.proofRequest,
+                stepAction: stepAction.id,
+              }
+            }),
+          )
+          .returning()
+
+        return {
+          ...stepResult,
+          actions: actionsResult.map((action) => ({
             ...action,
-            step: stepResult.id,
+            proofRequest: proofRequestsResult.find((proofRequest) => proofRequest.stepAction === action.id),
```

Comment:
I didn't get this line

Reply:
The code comes from development. The second return statement is mine